src/core/lib/surface/completion_queue.c

```diff
@@ -592,6 +598,7 @@ grpc_event grpc_completion_queue_next(grpc_completion_queue *cc,
       gpr_mu_lock(cc->mu);
       continue;
     } else {
+      gpr_atm_no_barrier_fetch_add(&cc->num_polls, 1);
```

Comment:
Agreed. We're already holding the lock at this point, and we are going to call this increment way more than we ever call the "get" function. So better just to make this an int and then put a lock there.